### PR TITLE
112146675 SEO

### PR DESCRIPTION
**Pull request by lollypop27**
SEO

Link to PT story: 
https://www.pivotaltracker.com/story/show/112146675

Description of the PR:
1. Page Title Tags for each Organisation
2. Meta descriptions for each Organisation
3. Header Tag and paragraph add to home page

### CODE REVIEW

**Comment by tansaku** [comment]
@lollypop27 thanks for this - looking great.

I think we might just have some merge conflicts in one of the fixture files - will see if I can fix that up for you

**Comment by tansaku** [comment]
@lollypop27 so I did some work here to sort things out.

I think I can infer some of the challenges you had with git which arise from keeping your fork in sync.

I've cherry-picked out your work and added it on top of the latest develop and then forced pushed into your branch here.  Apologies that's a little heavy handed, but I think it will make things clearer and I'm very happy to go through everything I did with you.

git flow can be a real pain.  You've got some great work in here, and I think now that I've adjusted things we'll be able to focus on that.

You will need to delete your branch locally and re-pull if you want to do further work - hope we can coordinate :-)

**Comment by lollypop27** [comment]
@tansaku Ok perfect thank you, I will re pull and go through it and we can chat

**Comment by tansaku** [comment]
hmm, I wonder if I dropped a commit ... not seeing changes to the application controller.

For reference it looks like we're working against this gem: https://github.com/kpumuk/meta-tags

File: app/controllers/organisations_controller.rb [hunk 1]
```diff
@@ -37,14 +37,23 @@ def show
     @can_propose_edits = current_user.present? && !@editable
     @markers = build_map_markers(organisations)
     @cat_name_ids = Category.name_and_id_for_what_who_and_how
+    set_meta_tags title: 'org.name',
```
Comment (tansaku) [inline app/controllers/organisations_controller.rb]: here, should we have `title: @organisation.name` ?

Comment (lollypop27) [inline app/controllers/organisations_controller.rb]: @tansaku yes I just used org.name because that was what was in the show page to keep it consistent, but is as far as I got.

**Comment by tansaku** [comment]
@lollypop27 this is looking great!

I have two thoughts.

one is that we might just move the meta_tags thing out into a separate private method in the organisations controller

The other is that we don't have any tests in place for this.  I'm thinking maybe we could have a couple of cucumber tests.  What do you think?

**Comment by lollypop27** [comment]
@tansaku can you please check and see if my testing process is ok and if I am done or need to test anything else?

**Comment by tansaku** [comment]
@lollypop27 this is looking good - there are a few tweaks and changes I might make.  Would you like me to give you the details so you can make the changes yourself, or would it be better if I just pushed them up and you reviewed the changes, and I could give extra explanation where you wanted it?

File: app/controllers/organisations_controller.rb [hunk 2]
```diff
@@ -37,14 +37,26 @@ def show
     @can_propose_edits = current_user.present? && !@editable
     @markers = build_map_markers(organisations)
     @cat_name_ids = Category.name_and_id_for_what_who_and_how
+    set_meta_tags title: @organisation.name,
+                  site: 'Harrow Community Network | Harrow volunteering',
+                  reverse: true,
+                  description: @organisation.description,
+                  author: 'http://www.agileventures.org',
+                  og: {
```
Comment (johnnymo87) [inline app/controllers/organisations_controller.rb]: What does `og` mean?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -2,18 +2,10 @@
 <html>
 <head>
   <meta http-equiv="content-type" content="text/html; charset=utf-8" />
-  <meta name="description" content="Harrow Community Network is a service provided by Voluntary Action Harrow to help people find voluntary services in their local community" />
-  <meta name="keywords" content="Harrow, Community, Network, volunteer, volunteering, voluntary, organisations, causes" />
   <meta name="google-site-verification" content="NnSd82Tg3wYtm75gYpDTcnfe5HzS5T38cU7oPDRlTD0" />
-
+  <%= display_meta_tags %>
```
Comment (johnnymo87) [inline app/views/layouts/application.html.erb]: Does `set_meta_tags` need to be used in the `ApplicationController`?

File: app/controllers/organisations_controller.rb (hunk 2 above)
Comment (diraulo) [inline app/controllers/organisations_controller.rb]: `og` = open graph

[more on Facebook Open Graph](https://developers.facebook.com/docs/sharing/webmasters)

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -568,3 +568,11 @@
   expect(stderr).not_to include "Error"
   expect(status).to be_success
 end
+
+Then(/^I should have a page with a title: "([^"]*)"$/) do |title|
+  expect(page).to have_title 'Harrow Community Network'
```
Comment (diraulo) [inline features/step_definitions/basic_steps.rb]: @lollypop27 you're hard coding the title's value in this test instead of using the one you extracted in `title`

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -568,3 +568,11 @@
...
+end
+
+Then(/^I should have a page with a description: "([^"]*)"$/) do |description|
+  expect(page).to have_description 'description'
```
Comment (diraulo) [inline features/step_definitions/basic_steps.rb]: @lollypop27 same here you want to ensure that the content of `description` actually gets rendered on the page and not `'description'`

**Comment by tansaku** [comment]
@lollypop27 put some review items in https://github.com/AgileVentures/LocalSupport/pull/371

File: features/step_definitions/basic_steps.rb (hunk 5 above)
Comment (lollypop27) [inline features/step_definitions/basic_steps.rb]: @diraulo ok thank you will update

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (lollypop27) [inline features/step_definitions/basic_steps.rb]: @diraulo ok thanks for that will update

**Comment by lollypop27** [comment]
@tansaku once checks are ready, please check and merge if happy

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (tansaku) [inline app/views/layouts/application.html.erb]: I think we do need it there @johnnymo87 but we'll do that in another PR

File: features/step_definitions/basic_steps.rb [hunk 6]
```diff
@@ -568,3 +568,11 @@
   expect(stderr).not_to include "Error"
   expect(status).to be_success
 end
+
+Then(/^I should have a page with a title: "([^"]*)"$/) do |title|
```
Comment (tansaku) [inline features/step_definitions/basic_steps.rb]: I guess we could adjust this to indicate that we mean a title tag in the head of the html document, so as not to confuse it with displaying an h1 title ...

Comment (lollypop27) [inline features/step_definitions/basic_steps.rb]: @tansaku ok will make the change and submit shortly

Comment (tansaku) [inline features/step_definitions/basic_steps.rb]: @lollypop27 feel free to do in next PR - was thinking out loud
